File: LeetCode/Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> parentheses_stack;

        for (int i = 0; i < s.length(); ++i){
            switch (s[i]){
                case '(':
                case '[':
                case '{':
                    parentheses_stack.push(s[i]);
                    break;
                case ')':
                    if (!parentheses_stack.empty() && parentheses_stack.top() == '('){
                        parentheses_stack.pop();
                    }else{
                        return false;
                    }
                    break;
                case ']':
                    if (!parentheses_stack.empty() && parentheses_stack.top() == '['){
                        parentheses_stack.pop();
                    }else{
                        return false;
                    }
                    break;
                case '}':
                    if (!parentheses_stack.empty() && parentheses_stack.top() == '{'){
                        parentheses_stack.pop();
                    }else{
                        return false;
                    }
                    break;
                default:
                    return false;
            }
        }
        if (parentheses_stack.empty()){
            return true;
        }else{
            return false;
        }
    }
};

File: LeetCode/Valid Parentheses.py
class Solution:
	# @return a boolean
	def isValid(self, s):
		stack = []

		for i in range(0, len(s)):
			ch = s[i]
			if ch == "(" or ch == "[" or ch == "{":
				stack.append(ch)
			elif ch == ")":
				if not stack or stack[-1] != "(":
					return False
				else:
					stack.pop()
			elif ch == "]":
				if not stack or stack[-1] != "[":
					return False
				else:
					stack.pop()				
			elif ch == "}":
				if not stack or stack[-1] != "{":
					return False
				else:
					stack.pop()
		if stack:
			return False
		return True

if __name__ == '__main__':
	s = Solution()
	st = '{}'
	print s.isValid(st)